core/runtime/executor.hpp

```diff
@@ -205,9 +204,9 @@ namespace kagome::runtime {
       BOOST_ASSERT_MSG(
           env.storage_provider->tryGetPersistentBatch(),
           "Current batch should always be persistent for a persistent call");
-      OUTCOME_TRY(
-          new_state_root,
-          env.storage_provider->tryGetPersistentBatch().value()->commit());
+      auto persistent_batch =
```

Comment:
hopefully neither result success nor persistent_batch allocation is guaranteed?

Reply:
commitState should only be called on persistent states, it's totally controlled by our code, so if it's called on ephemeral state, the node is supposed to crash, something went very wrong.

Reply:
We've checked it above with BOOST_ASSERT_MSG. It will not fail here.